test/network.spec.js

```diff
@@ -378,6 +378,33 @@ describe('Network Events', function() {
     expect(redirectedFrom.redirectedFrom()).toBe(null);
     expect(redirectedFrom.redirectedTo()).toBe(response.request());
   });
+  it('redirected fields should be available early', async({page, server}) => {
+    const expectRedirects = request => {
+      if (request.url().includes('foo.html')) {
+        expect(request.redirectedFrom()).toBe(null);
```

Comment:
I'd rather not have expects as event listener callbacks - they'll yield playwright in an unexpected state and might not be attributed to test.

Reply:
Done.